activesupport/lib/active_support/i18n_railtie.rb

```diff
@@ -55,7 +55,7 @@ def self.initialize_i18n(app)
 
       reloader = ActiveSupport::FileUpdateChecker.new(I18n.load_path.dup){ I18n.reload! }
       app.reloaders << reloader
-      ActionDispatch::Reloader.to_prepare { reloader.execute_if_updated }
+      ActionDispatch::Reloader.to_prepare { reloader.execute_if_updated.presence }
```

Comment:
> I'm pretty sure we are expecting prepare callbacks to stop if returning false. 

I didn't understand this sentence. In the current Rails code, ActionDispatch callbacks do not halt the chain when returning `false`. That setting is currently only present in [ActiveModel](https://github.com/rails/rails/pull/17227/files#diff-174733f2db65ef1bc53e3222d6ac0e61L20), in [ActiveModel::Validations](https://github.com/rails/rails/pull/17227/files#diff-0cd7b1cc8e754e9ab96c72b45770767dL26), and in [ActiveRecord](https://github.com/rails/rails/pull/17227/files#diff-174733f2db65ef1bc53e3222d6ac0e61L20).

Since currently we **don't** expect ActionDispatch callbacks to halt upon returning `false`, and the new default terminator **will** halt upon returning `false` and display a warning (unless `config.active_support.halt_callback_chains_on_return_false` is set to `false`), then I would add the following *custom* terminators for `#to_prepare` and `#to_cleanup`, so those callbacks will keep on **not halting** upon returning `false`:

```ruby
define_callbacks :prepare, terminator: ->(_, result_lambda) { result_lambda.call if result_lambda.is_a?(Proc); false }
define_callbacks :cleanup, terminator: ->(_, result_lambda) { result_lambda.call if result_lambda.is_a?(Proc); false }
```

Makes sense?